src/api/children.js

```diff
@@ -7,19 +7,12 @@ define(function () {
       return node;
     }
 
-    var treeWalker = document.createTreeWalker(node, NodeFilter.SHOW_ALL, null, false);
-    var previousNode = treeWalker.currentNode;
-    if (treeWalker.firstChild()) {
-      // TODO: build list of non-empty elements (used elsewhere)
-      // Do not include non-empty elements
-      if (treeWalker.currentNode.nodeName === 'BR') {
-        return previousNode;
-      } else {
-        return firstDeepestChild(treeWalker.currentNode);
-      }
-    } else {
-      return treeWalker.currentNode;
+    var child = node.firstChild;
+    if( child.nodeName === 'BR' ) {
+      return node;
     }
+
+    return firstDeepestChild(child);
```

Comment:
As far as my tests are concerned, NodeIterator and TreeWalker both perform really slow compared to Node or Selectors, that's why I ruled them out. But you are right, recursion might be slower than a loop—I don't know if browser engines optimize tail-recursive calls. At this point it might actually depend on the engine: http://jsperf.com/tail-recursive-vs-loop